lib/src/git_backend.rs

```diff
@@ -644,7 +639,25 @@ fn commit_from_git_without_root_parent(
     })
 }
 
-fn change_id_from_git_commit_id(id: &CommitId) -> ChangeId {
+/// Extracts change id from commit headers. If this fails, returns `Err` with an
+/// id from [`synthetic_change_id_from_git_commit_id`].
+pub fn extract_change_id_from_commit(
+    commit: &gix::objs::CommitRef,
+    id: &CommitId,
+) -> Result<ChangeId, ChangeId> {
```

Comment:
I think this function can be `(commit) -> Option<ChangeId>`. It's surprising that "extract" function generates new synthetic change id.

Reply:
Sure, SGTM, thanks!

Reply:
also update the commit mesage to match the new API

Reply:
Good point, thanks, done!